qutip/wigner.py

```diff
@@ -440,15 +448,15 @@ def qfunc(state, xvec, yvec, g=sqrt(2)):
     qmat = zeros(size(amat))
 
     if isket(state):
-        qmat = _qfunc_pure(state, amat)
+        qmat = _qfunc_pure(state, amat, amat_pwr)
     elif isoper(state):
         d, v = la.eig(state.full())
         # d[i]   = eigenvalue i
         # v[:,i] = eigenvector i
 
         qmat = zeros(np.shape(amat))
         for k in arange(0, len(d)):
-            qmat1 = _qfunc_pure(v[:, k], amat)
+            qmat1 = _qfunc_pure(v[:, k], amat, amat_pwr)
```

Comment:
Why not compute amat_pwr before the loop for large enough state. 
From my tests for 10x10 computing amat_pwr first is already faster.